GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2330,26 +2330,19 @@ private void CommitMessageToolStripMenuItemDropDownOpening(object sender, EventA
 
             void AddCommitMessageToMenu(string commitMessage)
             {
-                const int maxLabelLength = 50;
+                const int maxLabelLength = 72;
 
-                string label;
+                string label = commitMessage;
+                var newlineIndex = label.IndexOf('\n');
 
-                if (commitMessage.Length <= maxLabelLength)
+                if (newlineIndex != -1)
                 {
-                    label = commitMessage;
+                    label = label.Substring(0, newlineIndex);
                 }
-                else
-                {
-                    var newlineIndex = commitMessage.IndexOf('\n');
 
-                    if (newlineIndex != -1 && newlineIndex <= maxLabelLength)
-                    {
-                        label = commitMessage.Substring(0, newlineIndex);
-                    }
-                    else
-                    {
-                        label = commitMessage.ShortenTo(maxLabelLength);
-                    }
+                if (label.Length > maxLabelLength)
+                {
+                    label = label.ShortenTo(maxLabelLength);
```

Comment:
This could make a second allocation. Should that be allowed?

Reply:
It's not the RevisionGrid but only on user interaction.

Reply:
We preferred readability over performance because there is no performance constraints here.

Reply:
I don't think we have `Span` available in 4.6.1, so leave it as is